specs/switch-driver-capabilities.md

```diff
@@ -0,0 +1,51 @@
+
+# Expose network driver capabilites
+-----------
+
+[Issue 410](https://github.com/CCI-MOC/hil/issues/410)
+
+We want to expose switch driver capabilities. Capabilities could include:
+
+- Jumbo frames are per-switch on our Dell switches and per-port on our Ciscos (#755)
+-  QoS (#384)
+- Disabling STP (#779)
+- DHCP/ARP Snooping (#471)
+- Enabling the tagging of DHCP packets with the port number (from @okrieg or @pjd-nu)
+
+# Solution
+-----------
+
+* Every switch driver will have a list called `capabilities` in the class. So it would look like
+`capabilities = ["require-native-network", "per-port-jumbo-frames"]`
+
+* A new column (text) will be added to the switch table. This will store list of
+capabilities encoded as JSON. This could probably default to all capabilities as
+specified in the Class.
```

Comment:
If we don't support a way for modifying the capabilities per-switch yet, why not just keep this in the driver class and out of the DB?

Reply:
Yeah right. The question was should we support per-switch yet or not. If not, then we can just keep the capabilities in the class.

Reply:
Ya. And once we got per-switch, it might make sense to have a separate table since the relation is many to many.

Reply:
I think you misunderstood what I was proposing. The column is just one possible implementation strategy. The point is that, if we limit the interface to `get_capabilities`, the driver can come up with the list however it wants, whereas if we have a static attribute on the class, it's stuck leaving it the same for all switches. For an individual driver, it may not actualy make sense to store any extra information.

Also, the notion was to add a column to the *driver's* table, e.g. if the powerconnect driver wanted to track this information, it would add a column to a table defined in that extension. The table defined in hil.model doesn't change.